lib/JSON/Validator/OpenAPI/Mojolicious.pm

```diff
@@ -19,6 +19,24 @@ has version => 2;
 
 sub E { JSON::Validator::Error->new(@_) }
 
+has generate_definitions_path => sub {
+  my $self = shift;
+  Scalar::Util::weaken($self);
+
+  return sub {
+    my $ref = shift;
+
+    # Try to determine the path from the fqn
+    # We are only interested in the path in the fqn, so following fqn:
+    #
+    # #/components/schemas/some_schema, the returned path with be ['components', 'schemas']   (v3)
+    # #/definitions/some_schema, the returned path with be ['definitions']  (v2)
+
+    my $path = Mojo::Path->new( $ref->fqn =~ m!^.*#/(.+)$! )->to_dir->parts;
```

Comment:
This line will be overwritten next time I run perltidy.

```
perltidy -b lib/JSON/Validator/OpenAPI/Mojolicious.pm
```

Reply:
I don't think this will work though... What happens if the "$ref" points to "../generic-stuff.json#user"? Won't that put the definition inside "user" instead of "definitions" ? I think we need to whitelist what "$path" can and can't start with.

Please let me know if I've misunderstood something.

(Need to write more tests)

Reply:
`../generic-stuff.json#user` will not contain a path, so it will return definitions.

Could make this conditional on version, so only v3 is handled differently, and for v2 `['definitions']` is returned.
Using file references is one of the things that is not handled correctly (yet) for v3, not using it myself yet, but this part is mentioned in the OpenAPI3 guide :)

Tests are always a good thing, I will try to make a few. This could help to determine if this should be  handled differently for v2 and v3

Reply:
> This line will be overwritten next time I run perltidy.
> 
> ```
> perltidy -b lib/JSON/Validator/OpenAPI/Mojolicious.pm
> ```

Sorry about that, will up my game with next push :)

Reply:
> Could make this conditional on version, so only v3 is handled differently

That doesn't matter, since there's a hack in OpenAPI to ignore invalid $refs, and I'm pretty sure that flag works for v3 as well. I think we need two levels:

* If v2
  * if $path->[0] is "definitions", "parameters", "responses": Return $path ?
  * else: prefix with "definitions"?
* If v3
  * if $path->[0] is "components": Return $path ?
  * else prefix with "components"?

(I'm fine if you just want to fix it for 3x and always return "definitions" for 2x)

Reply:
> (I'm fine if you just want to fix it for 3x and always return "definitions" for 2x)

Its a safe bet to to this :)

Reply:
> > Could make this conditional on version, so only v3 is handled differently
> * If v3
>   
>   * if $path->[0] is "components": Return $path ?
>   * else prefix with "components"?

For v3, we still need to break it down, since /components contains sub paths for the different types of objects; `schemas`, `repsonses` and so on (reference: https://swagger.io/specification/#componentsObject)

Reply:
Wonder if the latest commit is hitting the mark, I have not found a good way to do testing for this